File: Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        int len = digits.size();

        if( len == 0 ){
            digits[0] = 1;
            return digits;
        }

        int i;
        for( i = len-1; i >= 0; i-- ){
            if( digits[i] == 9 ){
                digits[i] = 0;
            }else{
                digits[i]++;
                return digits;
            }
        }

        if(i < 0){
            digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};

File: LeetCode-Java/Plus One.java




public class Solution {
	public int[] plusOne(int[] digits) {
		int len = digits.length;
		int [] ret = new int [len+1];

		int carry = 1;
		for( int i = len ; i >= 1; i-- ){
			ret[i] = digits[i-1] + carry;
			carry = ret[i] / 10; 
			ret[i] %= 10;
		}
		ret[0] = carry;
		if( ret[0] == 0 ){
			ret = Arrays.copyOfRange( ret, 1, len+1 );
		}

		return ret; 
	}
}


